esmvaltool/cmorizers/obs/utilities.py

```diff
@@ -183,23 +183,33 @@ def save_variable(cube, var, outdir, attrs, **kwargs):
     iris.save(cube, file_path, fill_value=1e20, **kwargs)
 
 
-def extract_doi_value(tag):
+def extract_doi_value(tags):
     """Extract doi from a bibtex entry."""
-    reference_doi = 'doi not found'
+    reference_doi = []
     pattern = r'doi\ = {(.*?)\},'
 
-    bibtex_file = REFERENCES_PATH / f'{tag}.bibtex'
-    if bibtex_file.is_file():
-        reference_entry = bibtex_file.read_text()
-        if re.search("doi", reference_entry):
-            reference_doi = (
-                f'doi:{re.search(pattern, reference_entry).group(1)}'
+    if not isinstance(tags, list):
+        tags = [tags]
+
+    for tag in tags:
+        bibtex_file = REFERENCES_PATH / f'{tag}.bibtex'
+        if bibtex_file.is_file():
+            reference_entry = bibtex_file.read_text()
+            if re.search("doi", reference_entry):
+                reference_doi.append(
+                    f'doi:{re.search(pattern, reference_entry).group(1)}'
+                )
+            else:
+                reference_doi.append('doi not found')
+                logger.warning(
+                    'The reference file %s does not have a doi.', bibtex_file
+                )
+        else:
+            reference_doi.append('doi not found')
+            logger.warning(
+                'The reference file %s does not exist.', bibtex_file
```

Comment:
I'm just wondering if we can do this to get rid of the double failure condition:

```suggestion
    success = False
    if bibtex_file.is_file():
        reference_entry = bibtex_file.read_text()
        if re.search("doi", reference_entry):
            reference_doi.append(
                f'doi:{re.search(pattern, reference_entry).group(1)}'
            )
            success = True

    if not success:
        reference_doi.append('doi not found')
        logger.warning(
            'The reference file %s does not exist.', bibtex_file
```

Reply:
@stefsmeets thank you for the suggestion. Please note the failure and logger messages are not the same. The first failure is when there is not any doi in the bibtex file whereas the second indicates there is no a bibtex file for the tag. I think that clear logger messages are useful, what do you think?

Reply:
Oh, I completely looked over that. I agree with you, I just think this function got a bit too complicated for what it does.